docs/architecture/ADR-002-universal-share-encoding.md

```diff
@@ -13,36 +13,36 @@
 **reserved**: is the location of the first transaction, ISR, or evidence in this share if there is one and `0` if there isn't one
 **message length**: is the length of the entire message in bytes
 
-The current contiguous (transaction, ISRs, evidence) share format is:<br>`nid (8 bytes) | reserved (1 byte) | share data`
+The current reserved namespace (transaction, ISRs, evidence) share format is:<br>`nid (8 bytes) | reserved (1 byte) | share data`
```

Comment:
:+1:

Reply:
![Screenshot from 2022-08-25 16-18-38](https://user-images.githubusercontent.com/42654277/186770786-f6dc7d76-d649-4cc3-b548-9d5b75369504.png)

Reply:
after sleeping on this, while I really liked it at first, contiguous seems more descriptive as opposed to describing an implementation detail (reserved namespace). Also, is there anything stopping us from using a reserved namespace in a non-contiguous way (doesn't need the reserved byte)?

Reply:
> Also, is there anything stopping us from using a reserved namespace in a non-contiguous way

I don't think so besides updating the spec and implementation.

Agreed, "reserved" vs. "unreserved" is an implementation detail. My rationale for renaming from contiguous shares to reserved namespace shares is to disambiguate this reference from the separate but related term contiguous message shares. If possible, I think we should avoid using "contiguous" for both types of things. Message shares can be contiguous in that they are split across multiple shares but all part of one message. One alternative naming ideas for reserved vs. unreserved is non-message shares vs. message shares but that's also an implementation detail.

We want to use a word that describes how:
- one type of share can contain multiple things (transactions, ISRs, evidence) 
- one type of share can only contain 0 or 1 of a thing (messages)

I'm open to ideas

Reply:
how about monogamous shares vs polygamous shares? mono-shares poly-shares?

Reply:
- mono-content share vs poly-content shares?
- atomic-content share vs compound-content shares?

I don't like any of these as much as reserved vs non-reserved shares because readers now need to learn that these new terms map to the existing reserved vs non-reserved shares. In the future, if we introduce a reserved namespace that doesn't conform to the proposed share format (i.e. it doesn't support contiguous shares) then we can revise our terminology then, similar to how we introduced the notion of term: "universal share encoding"

Reply:
> We want to use a word that describes how:
one type of share can contain multiple things (transactions, ISRs, evidence)
one type of share can only contain 0 or 1 of a thing (messages)

would prefer something that does this.

personally, I don't like that we're using the same word to represent adjacent but different things (reserved namespaces and reserved byte). I can see that being confusing to new readers. I'd also prefer to avoid having to change names in the future.  Again, I feel like that will only add to the confusion of future readers.

since this is an opinionated thing that we don't have data for and could very well be a premature optimization, its not worth bike-shedding any further imo, and I'll definitely defer to the implementer's preference and other reviewers :slightly_smiling_face:

Reply:
Apologies for the several rounds of feedback on this issue. I think it's important for us to agree on a term to describe this. 

> I think that using the same word to represent adjacent but different things (reserved namespaces and reserved byte) could be confusing to new readers.

Thanks for articulating, I can now also see this being confusing to new readers. I think reserved namespace shares is superior to contiguous shares but still open to alternatives. One more suggestion that comes to mind:

- compact shares vs sparse shares

> compact: joined or packed together; closely and firmly united; [dense](https://www.dictionary.com/browse/dense); [solid](https://www.dictionary.com/browse/solid):

Source: https://www.dictionary.com/browse/compact

I think @adlerjohn is precise with language so I'm curious if he has opinions on this

Reply:
compact shares vs sparse shares is good

Reply:
Resolved in https://github.com/celestiaorg/celestia-app/pull/660/commits/c9a2fd52a04df1302a236c7efb613e2d9a6d28fc

Reply:
This discussion motivated https://github.com/celestiaorg/celestia-app/issues/688